akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickOffsetStore.scala

```diff
@@ -27,22 +27,37 @@ import slick.jdbc.JdbcProfile
   import OffsetSerialization.toStorageRepresentation
   import profile.api._
 
-  def this(db: P#Backend#Database, profile: P) = this(db, profile, Clock.systemUTC())
+  def this(db: P#Backend#Database, profile: P) =
+    this(db, profile, Clock.systemUTC())
 
   def readOffset[Offset](projectionId: ProjectionId)(implicit ec: ExecutionContext): Future[Option[Offset]] = {
     val action =
-      offsetTable.filter(_.projectionId === projectionId.id).result.headOption.map { maybeRow =>
-        maybeRow.map(row => fromStorageRepresentation[Offset](row.offsetStr, row.manifest))
+      offsetTable.filter(_.projectionId === projectionId.id).result.map { maybeRow =>
```

Comment:
The `ProjectionId.id` is unique for each Projection instance. For the MergableOffset case we said that it should read all rows for the same Projection type. We can store the `ProjectionId.name` and select all rows with the `name`.

We can probably figure out via these queries that it's a MergableOffset or if it's an ordinary offset, but if it's an ordinary offset it should only be the row with the right `projectionId.id`.

Reply:
Yes. To be clear, you're suggesting changing the query to return all rows for a `ProjectionId.name` and then figure out the offset type?

Reply:
I implemented this.